src/storage/local.rs

```diff
@@ -0,0 +1,85 @@
+#![allow(missing_docs)]
+
+use std::fs;
+use std::vec::Vec;
+use std::path::{Path, PathBuf};
+
+use core::{LalResult,config_dir,ensure_dir_exists_fresh};
+
+
+#[derive(Serialize, Deserialize, Clone, Default)]
+pub struct LocalConfig {}
+
+use super::{Backend, Component};
+
+pub struct LocalBackend {
+    /// Local config
+    pub config: LocalConfig,
+    /// Cache directory
+    pub cache: String,
+}
+
+impl LocalBackend {
+    pub fn new(cfg: &LocalConfig, cache: &str) -> Self {
+        LocalBackend{
+            config: cfg.clone(),
+            cache: cache.into(),
+        }
+    }
+}
+
+/// Artifact backend trait for `LocalBackend`
+///
+/// This is intended to be used by the caching trait `CachedBackend`, but for
+/// specific low-level use cases, these methods can be used directly.
+impl Backend for LocalBackend {
+    fn get_versions(&self, _name: &str, _loc: Option<&str>) -> LalResult<Vec<u32>> {
+        Ok(vec![0])
+    }
+
+    fn get_latest_version(&self, _name: &str, _loc: Option<&str>) -> LalResult<u32> {
+        Ok(0)
+    }
+
+    fn get_tarball_url(
+        &self,
+        name: &str,
+        _version: Option<u32>,
+        _loc: Option<&str>,
```

Comment:
What's a `loc`?

Reply:
yeah, env gets translated to loc in download.rs so could be renamed, but in storage land in means the bucket location so maybe sensible